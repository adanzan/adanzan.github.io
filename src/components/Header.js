import * as React from "react";

import { Button } from "@mui/material";

export default function Header () {
  return (
        <div>
            <Button variant="contained" disabled={true}>Download Resume</Button>
            {/* Call to action -> View resume */}
        </div>
  );
}
