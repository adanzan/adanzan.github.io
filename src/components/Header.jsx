/* eslint no-unused-vars: 0 */
import * as React from "react";

import { AppBar, Toolbar, Button } from "@mui/material";
import { Link, animateScroll as scroll } from "react-scroll";
import styles from "../styles/Header.module.css";

import logo from "../resources/images/logo.png";

export default function Header() {
	const SCROLL_DURATION = 500;
	const SCROLL_OFFSET = -80;
	const SCROLL_ANIMATION = "easeInOutCubic";

	return (
		<AppBar className={styles.appBar}>
			<Toolbar className={styles.toolbar} variant="regular">
				<Button onClick={() => scroll.scrollToTop()}>
					<div className={styles.logoWrapper}>
						<img className={styles.logo} src={logo} alt="logo" />
					</div>
				</Button>
				<div className={styles.buttonsWrapper}>
					<div className={styles.buttons}>
						<Link
							className={styles.link}
							activeClass={styles.linkActive}
							to="education"
							spy
							smooth={SCROLL_ANIMATION}
							offset={SCROLL_OFFSET}
							duration={SCROLL_DURATION}
						>
              EDUCATION
						</Link>

						<Link
							className={styles.link}
							activeClass={styles.linkActive}
							to="experience"
							spy
							smooth={SCROLL_ANIMATION}
							offset={SCROLL_OFFSET}
							duration={SCROLL_DURATION}
						>
              EXPERIENCE
						</Link>

						<Link
							className={styles.link}
							activeClass={styles.linkActive}
							to="skills"
							spy
							smooth={SCROLL_ANIMATION}
							offset={SCROLL_OFFSET}
							duration={SCROLL_DURATION}
						>
              SKILLS
						</Link>
						
						<Link
							className={styles.link}
							activeClass={styles.linkActive}
							to="projects"
							spy
							smooth={SCROLL_ANIMATION}
							offset={SCROLL_OFFSET}
							duration={SCROLL_DURATION}
						>
              PROJECTS
						</Link>

						<Link
							className={styles.link}
							activeClass={styles.linkActive}
							to="contact"
							spy
							smooth={SCROLL_ANIMATION}
							offset={SCROLL_OFFSET}
							duration={SCROLL_DURATION}
						>
              CONTACT
						</Link>
					</div>
				</div>
			</Toolbar>
		</AppBar>
	);
}
