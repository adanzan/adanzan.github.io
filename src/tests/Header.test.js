/* eslint no-unused-vars: 0 */
import * as React from "react";
import { render } from "@testing-library/react";

import Header from "../components/Header"; 

let testRender;

describe("Hero component", () => {
	beforeEach(() => {
		testRender = render(<Header />);
	});
	test("matches snapshot", () => {
		expect(testRender).toMatchSnapshot();
	});
	describe("displays", () => {
		test("logo", () => {
			expect(testRender.getByRole("img", "logo"));
		});
		test("titles", () => {
			expect(testRender.getByText("EDUCATION"));
			expect(testRender.getByText("EXPERIENCE"));
			expect(testRender.getByText("SKILLS"));
			expect(testRender.getByText("CONTACT"));
		});
	});

});
