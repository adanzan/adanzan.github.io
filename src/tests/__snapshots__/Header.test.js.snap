// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed appBar mui-fixed css-1jrbdul-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <div
              class="logoWrapper"
            >
              <img
                alt="logo"
                class="logo"
                src="logo.png"
              />
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="buttonsWrapper"
          >
            <div
              class="buttons"
            >
              <a
                class="link"
              >
                EDUCATION
              </a>
              <a
                class="link"
              >
                EXPERIENCE
              </a>
              <a
                class="link"
              >
                PROJECTS
              </a>
              <a
                class="link"
              >
                SKILLS
              </a>
              <a
                class="link"
              >
                CONTACT
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed appBar mui-fixed css-1jrbdul-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-hyum1k-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="logoWrapper"
          >
            <img
              alt="logo"
              class="logo"
              src="logo.png"
            />
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="buttonsWrapper"
        >
          <div
            class="buttons"
          >
            <a
              class="link"
            >
              EDUCATION
            </a>
            <a
              class="link"
            >
              EXPERIENCE
            </a>
            <a
              class="link"
            >
              PROJECTS
            </a>
            <a
              class="link"
            >
              SKILLS
            </a>
            <a
              class="link"
            >
              CONTACT
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
